File: rental/src/main/java/io/axoniq/demo/bikerental/rental/ui/SimulatorConfigController.java
package io.axoniq.demo.bikerental.rental.ui;

import org.springframework.context.annotation.Profile;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@Profile("simulator")
@RestController("/simulator")
public class SimulatorConfigController {

    private final Simulator simulator;

    public SimulatorConfigController(Simulator simulator) {
        this.simulator = simulator;
    }

    @PostMapping("/inventoryGenerationConfig")
    public void updateInventoryGeneratorConfig(@RequestParam(value = "size") Integer size,
                                               @RequestParam(value = "bikeType") String bikeType) {
        this.simulator.updateInventoryCreationConfiguration(size, bikeType);
    }

    @PostMapping("/rentalGenerationConfig")
    public void updateRentalGeneratorConfig(
            @RequestParam(value = "rentalBikeType") String rentalBikeType,
            @RequestParam(value = "loops") Integer loops,
            @RequestParam(value = "concurrency", defaultValue = "1") int concurrency,
            @RequestParam(value = "abandonPaymentFactor", defaultValue = "100") int abandonPaymentFactor,
            @RequestParam(value = "delay", defaultValue = "0")int delay) {
        this.simulator.updateRentalGenerationConfiguration(rentalBikeType, loops,concurrency, abandonPaymentFactor, delay);
    }
}

File: rental/src/ui/simulator-config-controller.ts
import { Router } from 'express';
import { Simulator } from './simulator.js';

/** Lets the running simulator's inventory- and rental-generation parameters be reconfigured. */
export function simulatorConfigRouter(simulator: Simulator): Router {
  const router = Router();

  router.post('/inventoryGenerationConfig', (req, res) => {
    simulator.updateInventoryCreationConfiguration(Number(req.query.size), String(req.query.bikeType));
    res.status(200).end();
  });

  router.post('/rentalGenerationConfig', (req, res) => {
    simulator.updateRentalGenerationConfiguration(
      String(req.query.rentalBikeType),
      Number(req.query.loops),
      req.query.concurrency ? Number(req.query.concurrency) : 1,
      req.query.abandonPaymentFactor ? Number(req.query.abandonPaymentFactor) : 100,
      req.query.delay ? Number(req.query.delay) : 0,
    );
    res.status(200).end();
  });

  return router;
}
